packages/apex-node/src/streaming/streamingClient.ts

```diff
@@ -85,19 +90,35 @@ export class StreamingClient {
     });
 
     this.client.addExtension({
-      incoming: (
+      incoming: async (
         message: StreamMessage,
         callback: (message: StreamMessage) => void
       ) => {
         if (message && message.error) {
           if (message.channel === '/meta/handshake') {
-            this.client.disconnect();
+            this.disconnect();
             throw new Error(
               nls.localize('streamingHandshakeFail', message.error)
             );
           }
 
-          this.client.disconnect();
+          if (message.error === StreamingErrors.ERROR_AUTH_INVALID) {
```

Comment:
Can we add some comment here on why we are handling 401 and 403 errors this way? You have them well explained in the quip doc. Will be good to have a shorter comment here.

Reply:
Good advice, added.